### PR TITLE
Bootstrap arcade

### PR DESCRIPTION
**Pull request by ViktorHofer**
Adding the Arcade SDK, eng/common/, boilerplate files and enabling dependency flow.

Review by commits. Basically only commit 3 is interesting, everything else is just a copy or move. The yml is copied from Arcade with minor adjustments.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+indent_size = 2
+tab_width = 2
+
+[*.{cs}]
+indent_size = 4
+tab_width = 4
+
+[*.{*proj,xml,props,targets,tasks}]
+indent_size = 2
+tab_width = 2
```
Comment (stephentoub) [inline .editorconfig]: Can/should we use the same .editorconfig as is in dotnet/runtime?

File: .gitignore [hunk 2]
```diff
@@ -2,45 +2,40 @@ syntax: glob
...
-.idea/
+artifacts_stage_1/
 [Dd]ebug/
-[Dd]ebugPublic/
 [Rr]elease/
-[Rr]eleases/
-build/
-bld/
+x64/
+x86/ !eng/common/cross/x86/
```
Comment (stephentoub) [inline .gitignore]: Is the space here correct?

File: .gitignore [hunk 3]
```diff
@@ -2,45 +2,40 @@ syntax: glob
...
-cross/rootfs/
-cross/android-rootfs/
-
-# Visual Studio
+# Visual Studio 2015
 .vs/
 
+# Visual Studio 2015 Pre-CTP6
+*.sln.ide
+*.ide/
```
Comment (stephentoub) [inline .gitignore]: I guess it doesn't hurt to still have these, but is it relevant to anyone?

File: .editorconfig (hunk 1 above)
Comment (ViktorHofer) [reply .editorconfig]: I copied it from Arcade. We can use the runtime one.

File: .gitignore (hunk 2 above)
Comment (ViktorHofer) [reply .gitignore]: Same here, I copied the file from Arcade. I can use the runtime one here as well.

Comment (ViktorHofer) [reply .gitignore]: yes, the space here is correct.

File: eng/common/native/CommonLibrary.psm1 [hunk 4]
```diff
@@ -0,0 +1,389 @@
+<#
```
Comment (safern) [inline eng/common/native/CommonLibrary.psm1]: I don't think we need files like `eng/common/native`, `eng/common/internal`, `eng/common/performance`... should we add them to the .gitignore and remove them?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    exclude:
+    - docs/*
+    
+variables:
+- template: eng/common-variables.yml
+
+resources:
+  containers:
+  - container: LinuxContainer
+    image: microsoft/dotnet-buildtools-prereqs:ubuntu-14.04-cross-0cd4667-20170319080304
```
Comment (safern) [inline azure-pipelines.yml]: We should use a newer container and one that comes from mcr registry instead.

**Pull request merged**